dll/win32/shell32/dialogs/filedefext.cpp

```diff
@@ -299,25 +299,31 @@ SH_FormatFileSizeWithBytes(const PULARGE_INTEGER lpQwSize, LPWSTR pwszResult, UI
  */
 
 HPROPSHEETPAGE
-SH_CreatePropertySheetPage(WORD wDialogId, DLGPROC pfnDlgProc, LPARAM lParam, LPCWSTR pwszTitle)
+SH_CreatePropertySheetPageEx(WORD wDialogId, DLGPROC pfnDlgProc, LPARAM lParam,
+                             LPCWSTR pwszTitle, LPFNPSPCALLBACK Callback)
 {
-    PROPSHEETPAGEW Page;
-
-    memset(&Page, 0x0, sizeof(PROPSHEETPAGEW));
-    Page.dwSize = sizeof(PROPSHEETPAGEW);
-    Page.dwFlags = PSP_DEFAULT;
-    Page.hInstance = shell32_hInstance;
+    PROPSHEETPAGEW Page = { sizeof(Page), PSP_DEFAULT, shell32_hInstance };
     Page.pszTemplate = MAKEINTRESOURCE(wDialogId);
     Page.pfnDlgProc = pfnDlgProc;
     Page.lParam = lParam;
     Page.pszTitle = pwszTitle;
+    Page.pfnCallback = Callback;
 
     if (pwszTitle)
         Page.dwFlags |= PSP_USETITLE;
 
+    if (Callback)
+        Page.dwFlags |= PSP_USECALLBACK;
+
     return CreatePropertySheetPageW(&Page);
 }
 
+HPROPSHEETPAGE
+SH_CreatePropertySheetPage(WORD wDialogId, DLGPROC pfnDlgProc, LPARAM lParam, LPCWSTR pwszTitle)
```

Comment:
Shouldn't these two functions also be moved into the "core-functionality" propsheet.cpp file, following what @yagoulas suggested?